ng/src/web/pages/schedules/dialog.js

```diff
@@ -52,104 +64,151 @@ const TimeUnitSelect = props => {
+  if (is_defined(timezone)) {
+    data.timezone = timezone;
+  };
+
   return (
-    <Layout flex="column">
-
-      <FormGroup title={_('Name')}>
-        <TextField
-          name="name"
-          grow="1"
-          value={name}
-          size="30"
-          onChange={onValueChange}
-          maxLength="80"/>
-      </FormGroup>
-
-      <FormGroup title={_('Comment')}>
-        <TextField
-          name="comment"
-          value={comment}
-          grow="1"
-          size="30" maxLength="400"
-          onChange={onValueChange}/>
-      </FormGroup>
-
-      <FormGroup title={_('First Time')}>
-        <DatePicker
-          name="date"
-          value={date}
-          onChange={onValueChange}/>
-        <Divider>
-          <Spinner
-            name="hour"
-            type="int"
-            min="0"
-            max="23"
-            size="2"
-            value={hour}
-            onChange={onValueChange}/> h
-          <Spinner
-            name="minute"
-            type="int"
-            min="0"
-            max="59"
-            size="2"
-            value={minute}
-            onChange={onValueChange}/> m
-        </Divider>
-      </FormGroup>
-
-      <FormGroup title={_('Timezone')}>
-        <TimeZoneSelect
-          name="timezone"
-          value={timezone}
-          onChange={onValueChange}/>
-      </FormGroup>
-
-      <FormGroup title={_('Period')}>
-        <Divider>
-          <Spinner
-            name="period"
-            type="int"
-            min="0"
-            size="3"
-            value={period}
-            onChange={onValueChange}/>
-          <TimeUnitSelect
-            name="period_unit"
-            value={period_unit}
-            onChange={onValueChange}/>
-        </Divider>
-      </FormGroup>
-
-      <FormGroup title={_('Duration')}>
-        <Divider>
-          <Spinner
-            name="duration"
-            type="int"
-            min="0"
-            size="3"
-            value={duration}
-            onChange={onValueChange}/>
-          <TimeUnitSelect
-            name="duration_unit"
-            value={duration_unit}
-            onChange={onValueChange}/>
-        </Divider>
-      </FormGroup>
-    </Layout>
+    <SaveDialog
+      visible={visible}
+      title={title}
+      onClose={onClose}
+      onSave={onSave}
+      initialData={data}
+    >
+      {({
+        data: state,
+        onValueChange,
+      }) => {
+        return (
+          <Layout flex="column">
+
+            <FormGroup title={_('Name')}>
+              <TextField
+                name="name"
+                grow="1"
+                value={state.name}
+                size="30"
+                onChange={onValueChange}
+                maxLength="80"/>
+            </FormGroup>
+
+            <FormGroup title={_('Comment')}>
+              <TextField
+                name="comment"
+                value={state.comment}
+                grow="1"
+                size="30" maxLength="400"
```

Comment:
Coding style nitpick: move maxLength="400" on the next line.

Reply:
Done